### ISSUE TITLE
Tax calculation issue for country/state specific tax when inputing prices inclusive

### ISSUE THREAD

**Issue by jkohlbach** [woocommerce/woocommerce#15736]
## Prerequisites

- [ X ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [ X ] The issue still exists against the latest `master` branch of WooCommerce
- [ X ] This is not a usage question (Those should be directed to the [community](https://wordpress.org/support/plugin/woocommerce), unless this is a question about a premium plugin in which you should [use the helpdesk](https://woocommerce.com/my-account/tickets/) for official extensions or contact the author of 3rd party extensions)
- [ X ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

Hi guys, I found this was easier to explain via video:
https://www.dropbox.com/s/pe5zis3ybj7a509/wc3.1-tax-calculation-issue-state-based-taxes.mp4?dl=0

TL;DR
1. Add a product
2. Set setting to prices entered inclusive of tax
3. Set a country and state specific tax rate
4. When viewing the product on the front end it will add tax on top instead of as a portion of the total price of the product (inclusive).
5. To prove that this is weird, try changing the tax rate to global (meaning not restricted to a country or state but still the same tax rate). It then correctly presents the tax as a portion of the total price of the product as expected.

## Expected behavior and actual behavior

When I follow those steps, I see... the tax gets added on top of the price defined for the product.

I was expecting... it would be a portion of the total price of the product (as it correctly is when you remove the country and state code).

## Environment

Completely fresh WP 4.8 and WC 3.0.8 install (latest at time of writing), PHP7, MySQL 5.6. No extra plugins added, its 100% vanilla.

<details>
```
`
### WordPress Environment ###

Home URL: http://localhost:8888/test3
Site URL: http://localhost:8888/test3
WC Version: 3.0.8
Log Directory Writable: ✔
WP Version: 4.8
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.2.29 (Unix) mod_wsgi/3.5 Python/2.7.10 PHP/7.0.0 mod_ssl/2.2.29 OpenSSL/0.9.8zh DAV/2 mod_fastcgi/2.4.6 mod_perl/2.0.9 Perl/v5.22.0
PHP Version: 7.0.0
PHP Post Max Size: 32 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.43.0
OpenSSL/0.9.8zd

SUHOSIN Installed: –
MySQL Version: 5.6.35
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ❌ wp_remote_post() failed. Contact your hosting provider. cURL error 35: Unsupported SSL protocol version
Remote Get: ❌ wp_remote_get() failed. Contact your hosting provider. 200

### Database ###

WC Database Version: 3.0.8
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (2) ###

WooCommerce: by Automattic – 3.0.8
WordPress Importer: by wordpressdotorg – 0.6.3

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #103 - /shop/
Cart: #104 - /cart/
Checkout: #105 - /checkout/
My account: #106 - /my-account/

### Theme ###

Name: Twenty Seventeen
Version: 1.3
Author URL: https://wordpress.org/
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: –
`
```
</details>

## Isolating the problem

- [ X ] This bug happens with only WooCommerce plugin active
- [ X ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [ X ] I can reproduce this bug consistently

**Comment by mikejolley** [comment]
Tax is inclusive for the base location. If the tax rate does not match the base location exactly, it's added on top and the 'inclusive' amount of tax is assumed 0.

Your example worked my side with a base location of queensland. You did not show that in your video but I'm assuming it's something else :)

Thanks

**Comment by jkohlbach** [comment]
Hey Mike, yep thats exactly right, the base location was UK I think so that would explain.

However I still think the calculation is incorrect.... bear with me!

Let me try to explain with some examples :)

**First issue, the admin doesn't have a consistent view of what the "prices entered inclusive" setting really does in practice.**

The example I gave with the global tax I had specified the customer's location to "Australia" so the location was definitely outside of the base location, yet it made the tax inclusive on the amount (which I think is correct).

If I enter prices "inclusive" then store admin would likely assume the total price of the product shown to customers would be inclusive of any and all taxes defined.

This IS the case if we use a global tax, but is NOT the case if we use a specific location tax. We're basically ignoring the "prices entered inclusive" setting which could be very confusing for the store admin during setup.

**Second issue is from a business perspective it doesn't mesh well with my tax obligations.**

If I'm outside the taxed zone, then there should just be no tax component to the price (this works as expected).

But if the customer is inside the taxed zone, then it should make the tax a component of the price because the price was entered as inclusive of tax.

I'm struggling to see why the base location should have anything to do with it?

What I'm doing as a store owner is defining taxes for a specific location that I have an obligation to collect tax for. Where I have my business based is irrelevant because I might have an obligation to collect tax for another country as well as the country I'm based in. This is the same for state based sales taxes I'd think?

**Third issue I see is that for customers, its inconsistent.**

I can't ensure a consistent price for all locations (regardless of whether the store owner collects tax as part of that transaction). There is currently no way in the system to be able to collect tax across multiple zones all while presenting the same total price to anyone in the world.

Eg. If I had a business with locations in Australia, New Zealand and Canada, then I need to collect tax in all three locations and if I'm selling in USD then I may also want the price to be consistent across all those countries (as well as any other country that view it)

If the admin wants the tax on top, they could enter the prices exclusive of tax. That's what the setting is for, right?

Can you clarify why would it be different based on the store's zone vs. the buyer's zone? Is it maybe to account for tax laws like EU VAT? If so couldn't you just use a global tax as demonstrated?

Thanks and apologies if this is just me being dense and not getting it.

**Comment by mikejolley** [comment]
It works so that people outside of your base location do not pay your taxes.

See https://github.com/woocommerce/woocommerce/wiki/How-Taxes-Work-in-WooCommerce#prices-including-tax---experimental-behavior
